src/lib/containers/QueryContext.tsx

```diff
@@ -0,0 +1,125 @@
+import * as React from 'react'
+import { createContext, useContext } from 'react'
+import { SynapseClientError } from '../utils/SynapseClient'
+import {
+  AsynchronousJobStatus,
+  QueryBundleRequest,
+  QueryResultBundle,
+  Table,
+} from '../utils/synapseTypes'
+
+export const QUERY_FILTERS_EXPANDED_CSS: string = 'isShowingFacetFilters'
+export const QUERY_FILTERS_COLLAPSED_CSS: string = 'isHidingFacetFilters'
+
+/*
+  For details page: to lock a facet name (e.g. study, grant) so that the facet name
+  and its all possible values will not appear on the details page. The facet name is
+  given by the url's search param. The type is defined here so that other child components
+  in SRC won't generate type errors.
+ */
+export type LockedFacet = {
+  facet?: string
+  value?: string
+}
+
+export type QueryContextType = {
+  /** The entity being queried. Will be undefined while initially fetching */
+  entity: Table | undefined
+  /** The query results, which will be undefined while initially fetching a new bundle, but will not be unloaded when fetching new pages */
+  data: QueryResultBundle | undefined
+  /** Returns a deep clone of the current query bundle request */
+  getLastQueryRequest: () => QueryBundleRequest
+  /** Returns a deep clone of the initial query bundle request */
+  getInitQueryRequest: () => QueryBundleRequest
+  /** Updates the current query with the passed request */
+  executeQueryRequest: (param: QueryBundleRequest) => void
+  /** Returns true when loading a new page of query results */
+  isLoadingNewPage: boolean
+  /** Returns true when loading a brand new query result bundle. Will not be true when just loading the next page of query results. */
+  isLoadingNewBundle: boolean
+  /** The error returned by the query request, if one is encountered */
+  error: SynapseClientError | null
+  /** The status of the asynchronous job. */
+  asyncJobStatus?: AsynchronousJobStatus<QueryBundleRequest, QueryResultBundle>
+  /** Whether or not facets are available to be filtered upon based on the current data */
+  isFacetsAvailable: boolean
+  /**
+   * A facet may be "locked" so that it is not modifiable by the user, for example when showing only data relevant to a particular facet value on a Details Page.
+   * The value of a locked facet will result in a client-side modification of the result bundle data.
+   */
+  lockedFacet?: LockedFacet
+}
+
+export type PaginatedQueryContextType = QueryContextType & {
+  /** Navigates to a particular page */
```

Comment:
No preference for if you make this 0-indexed or 1-indexed, but either way it should be explicitly documented. Currently it's 1-indexed.
```suggestion
  /** Navigates to a particular page, where the first page has value 1 */
```